lib/httpServerFactory.js

```diff
@@ -1,14 +1,45 @@
 /* 
- * httpServerFactory.js, HTTP server factory
+ * httpServerFactory.js, HTTP & HTTPS server factory
  * 
  * (C) 2018 TekMonks. All rights reserved.
  */
 
-const http = require("http");
+const fs = require("fs");
 const urlMod = require("url");
-ESB.env.HTTPSERVERS = {};
+const http = require("http");
```

Comment:
The code is exact same for HTTP and HTTPS server creation except the SSL options object. So there should be no need for 2 functions which are essentially copies of each other. Needs to be a single function with maybe params from listener making the server factory decide whether to call http or https class. Similarly SSL object config should be coming from ESB flow not SSL.json. I can decide to create multiple HTTPS servers with DIFFERENT certificates for my flows if I wish. So rejecting this pull.

Principles of coding
(1) Do NOT REPEAT code - condition the branching if needed, not repeat entire code just for minor branching logic.
(2) Think about the implications of your design - your current design FORCES a single SSL config for the entire server - why? Can't I have multiple SSL servers on different ports with different SSL configs?